cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go

```diff
@@ -119,6 +127,61 @@ func (t *replicateDiskTask) GetResponse() proto.Message {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+func (t *replicateDiskTask) createProxyOverlayDisk(
+	ctx context.Context,
+	execCtx tasks.ExecutionContext,
+	client nbs_client.Client,
+	currentCheckpointID string,
+) (string, error) {
+
+	diskID := t.request.SrcDisk.DiskId
+	proxyOverlayDiskID := common.GetProxyOverlayDiskID(
+		diskID,
+		currentCheckpointID,
+	)
+
+	created, err := client.CreateProxyOverlayDisk(
+		ctx,
+		proxyOverlayDiskID,
+		diskID,
+		currentCheckpointID,
+	)
+	if err != nil {
+		return "", err
+	}
+
+	if created {
+		return proxyOverlayDiskID, nil
+	}
+
+	return "", nil
```

Comment:
Is this possible? What happens in this case?

Reply:
https://github.com/ydb-platform/nbs/blob/8e8f5d5b4b226051f356dd955b05f36d4b28de0e/cloud/disk_manager/internal/pkg/clients/nbs/client.go#L608

Reply:
everything will be ok:
https://github.com/ydb-platform/nbs/blob/main/cloud/disk_manager/internal/pkg/dataplane/nbs/source.go#L249